src/Kunstmaan/MenuBundle/Entity/BaseMenuItem.php

```diff
@@ -0,0 +1,418 @@
+    public function getTitle()
+    {
+        return $this->title;
+    }
+
+    /**
+     * @param string $title
+     *
+     * @return MenuItem
+     */
+    public function setTitle($title)
+    {
+        $this->title = $title;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getUrl()
+    {
+        return $this->url;
+    }
+
+    /**
+     * @param string $url
+     *
+     * @return MenuItem
+     */
+    public function setUrl($url)
+    {
+        $this->url = $url;
+
+        return $this;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function isNewWindow()
+    {
+        return $this->newWindow;
+    }
+
+    /**
+     * @param boolean $newWindow
+     *
+     * @return MenuItem
+     */
+    public function setNewWindow($newWindow)
+    {
+        $this->newWindow = $newWindow;
+
+        return $this;
+    }
+
+    /**
+     * @return int
+     */
+    public function getLft()
+    {
+        return $this->lft;
+    }
+
+    /**
+     * @param int $lft
+     *
+     * @return MenuItem
+     */
+    public function setLft($lft)
+    {
+        $this->lft = $lft;
+
+        return $this;
+    }
+
+    /**
+     * @return int
+     */
+    public function getLvl()
+    {
+        return $this->lvl;
+    }
+
+    /**
+     * @param int $lvl
+     *
+     * @return MenuItem
+     */
+    public function setLvl($lvl)
+    {
+        $this->lvl = $lvl;
+
+        return $this;
+    }
+
+    /**
+     * @return int
+     */
+    public function getRgt()
+    {
+        return $this->rgt;
+    }
+
+    /**
+     * @param int $rgt
+     *
+     * @return MenuItem
+     */
+    public function setRgt($rgt)
+    {
+        $this->rgt = $rgt;
+
+        return $this;
+    }
+
+    /**
+     * @return MenuItem
+     */
+    public function getParent()
+    {
+        return $this->parent;
+    }
+
+    /**
+     * @param MenuItem $parent
+     *
+     * @return MenuItem
+     */
+    public function setParent($parent)
+    {
+        $this->parent = $parent;
+
+        return $this;
+    }
+
+    /**
+     * @return ArrayCollection
+     */
+    public function getChildren()
+    {
+        return $this->children;
+    }
+
+    /**
+     * @param ArrayCollection $children
+     *
+     * @return MenuItem
+     */
+    public function setChildren($children)
+    {
+        $this->children = $children;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDisplayTitle()
+    {
+        if ($this->getType() == self::TYPE_PAGE_LINK) {
+            if (!is_null($this->getTitle())) {
+                return $this->getTitle();
+            } else {
+                return $this->getNodeTranslation()->getTitle();
+            }
+        } else {
+            return $this->getTitle();
+        }
+    }
+
+    /**
+     * @return string
+     */
+    public function getSelectTitle()
+    {
+        if ($this->getType() == self::TYPE_PAGE_LINK) {
```

Comment:
Why was this copy/pasted, it's exactly the same as getDisplayTitle so you should call that here instead...